test/text/test_render_styled.py

```diff
@@ -545,3 +545,26 @@ def test_multi_cell_ln_newpos(tmp_path):
             doc.line(doc.x, doc.y - 3, doc.x, doc.y + 3)
 
     assert_pdf_equal(doc, HERE / "multi_cell_ln_newpos.pdf", tmp_path)
+
+
+def test_cell_skew_text(tmp_path):
+    doc = fpdf.FPDF()
+    doc.add_page()
+    doc.set_font("helvetica", size=12)
+    doc.ln(40)
+    with doc.skew_text(0, 20):
+        doc.cell(txt="text skewed on the y-axis")
+        doc.ln(40)
+    with doc.skew_text(0, -20):
+        doc.cell(txt="text skewed on the y-axis (negative)")
+        doc.ln(40)
+    with doc.skew_text(20, 0):
+        doc.cell(txt="text skewed on the x-axis")
+        doc.ln(40)
+    with doc.skew_text(-20, 0):
+        doc.cell(txt="text skewed on the x-axis (negative)")
+        doc.ln(40)
+    with doc.skew_text(89, 0):
+        doc.cell(txt="some extreme skewing")
+        doc.ln(40)
+    assert_pdf_equal(doc, HERE / "cell_skew_text.pdf", tmp_path)
```

Comment:
Nice unit test!
Could you also provite an extra small test with `FPDF.multi_cell` and some input text covering at least 2 lines?

Reply:
OK small problem. Just found out that `_render_styled_text_line` supports a `fill` option which generates rectangles around the text. They are graphics, and thus not affected by the `Tm` operator:  
![Screenshot from 2022-12-17 17-34-50](https://user-images.githubusercontent.com/30405025/208249467-fa8f00a3-129b-4e20-94de-76f2086acb05.png)
I thought of replacing it with a `cm` but then that would just be a `skew` function and not specifically `skew_text`.  
I am guessing that I can limit the skewing only to the rectangle around the text with a few conditions in the `_render_styled_text` function (I'll also have to solve a weird problem where setting a skew on the x-axis creates a location offset as well, but that's probably solvable): 
![Screenshot from 2022-12-17 17-32-59](https://user-images.githubusercontent.com/30405025/208249404-43c3a795-03af-4f10-b468-0ae21b922c76.png)  
Another option of course would be not to support the `fill` option together with skewed text, but that may be limiting this feature.
What do you suggest?

Reply:
I like the idea of a skew function that would apply to not only text.
If you want to follow that road, that is really fine with me, and may be even more useful to `fpdf2` users.

If you to stick to a `skew_text` method, you can just forbid it to be used with `fill=True` and raise a `FPDFException` or `ValueError` if it happens.

Reply:
OK, after playing with it a bit I found that the transformation needed to fix the unwanted movement and leave only the skew is this:  
```
        lim_val = 2**32
        x = self.x
        y = self.y
        ax = max(min(math.tan(ax * (math.pi / 180)), lim_val), -lim_val)
        ay = max(min(math.tan(ay * (math.pi / 180)), lim_val), -lim_val)
        cx, cy = x * self.k, (self.h - y) * self.k
        with self.local_context():
            self._out(
                f"1 {ay:.5f} {ax:.5f} 1 {cx:.2f} {cy:.2f} cm "
                f"1 0 0 1 -{cx:.2f} -{cy:.2f} cm"
            )
            yield
```
So I guess I can implement a `skew` function instead of `skew_text` and it will be conceptually very similar to `rotation`, what do you say?

Reply:
Sounds great @erap129!
Go on, that will be a nice addition to `fpdf2` 😊